### PR TITLE
Lwg 22 05 24

### PR DESCRIPTION
**Pull request by crtrott**
This implements all the notes made for the LWG small group meeting on 2022-05-24 noted here: #247 except adding swap

### CODE REVIEW

**Comment by crtrott** [comment]
I have NOT updated paragraph numbers yet in order to make review easier.

File: P0009/P0009.md [hunk 1]
```diff
@@ -2868,19 +2897,10 @@ template<class OtherElementType, class OtherExtents,
 
 [19]{.pnum} *Remarks:* The expression inside `explicit` is:
 ```c++
-  !is_convertible_v<const typename OtherLayoutPolicy::mapping_type&, mapping_type> ||
-  !is_convertible_v<const OtherAccessorPolicy&, AccessorPolicy>
-```
-
-```c++
-template <class ElementType, class... Integrals>
-explicit mdspan(ElementType*, Integrals...)
```
Comment (mhoemmen) [inline P0009/P0009.md:2877]: Did this deduction guide go away or does it just not belong here?

File: P0009/P0009.md [hunk 2]
```diff
@@ -2410,7 +2412,7 @@ template<class OtherMapping>
 
 [1]{.pnum} An *accessor policy* defines types and operations by which a reference to a single object is created from an abstract data handle to a number of such objects and an index.
 
-[2]{.pnum} A range of indicies $[0, N)$ is an *accessible range* of a given data handle and an accessor, if for each `i` in the range the accessor policies `access` function produces a valid reference to an object.
+[2]{.pnum} A range of indicies $[0, N)$ is an *accessible range* of a given data handle and an accessor, if for each `i` in the range the accessor policy's `access` function produces a valid reference to an object.
```
Comment (mhoemmen) [inline P0009/P0009.md:2415]:
```suggestion
[2]{.pnum} A range of indices $[0, N)$ is an *accessible range* of a given data handle and an accessor, if for each `i` in the range the accessor policy's `access` function produces a valid reference to an object.
```

**Review by mhoemmen — CHANGES_REQUESTED** [review]

File: P0009/P0009.md [hunk 3]
```diff
@@ -1620,7 +1620,9 @@ template <size_t Rank>
 
 [1]{.pnum} A type `M` meets the *layout mapping* requirements if: 
 
-   * [1.1]{.pnum} `M` models `regular`,
+   * [1.1]{.pnum} `M` models `copyable`,
+
+   * [1.2]{.pnum} `M` models `equality_comparable`,
```
Comment (dalg24) [inline P0009/P0009.md:1625]: You need to update the numbering below

File: P0009/P0009.md [hunk 4]
```diff
@@ -2575,12 +2577,7 @@ constexpr pointer offset(pointer p, size_t i) const noexcept;
 <br/>
 <b>24.7.�.1 Overview [mdspan.mdspan.overview]</b>
 
-[1]{.pnum} `mdspan` maps a multidimensional index in its domain
-   to a reference to an element in its codomain.
-
-[2]{.pnum} The *domain* of an `mdspan` object is a multidimensional index space defined by an `extents`.
-
-[3]{.pnum} The *codomain* of an `mdspan` object is a set of elements accessible from a contiguous range of integer indices.
+[1]{.pnum} `mdspan` is a view of an multidimensional array of elements.
```
Comment (dalg24) [inline P0009/P0009.md:2580]:
```suggestion
[1]{.pnum} `mdspan` is a view of a multidimensional array of elements. 
```

File: P0009/P0009.md [hunk 5]
```diff
@@ -2704,24 +2698,32 @@ mdspan(ElementType*, const MappingType&, const AccessorType&)
 }
 ```
 
-<!-- mfh 20 Jan 2019: Putting the template parameters after the class declaration imitates [span.overview]. -->
+[2]{.pnum} *Mandates:*
+
+   * [2.1]{.pnum} `ElementType` is a a complete object type that is neither an abstrct class type nor an array type,
```
Comment (dalg24) [inline P0009/P0009.md:2703]:
```suggestion
   * [2.1]{.pnum} `ElementType` is a a complete object type that is neither an abstract class type nor an array type,
```

File: P0009/P0009.md [hunk 6]
```diff
@@ -2758,6 +2782,8 @@ template<class... SizeTypes>
 
    * [3.2]{.pnum} Direct-non-list-initializes _`map_`_ with `extents_type(static_cast<size_type>(std::move(exts))...)`.
 
+   * [2.3]{.pnum} Value-initializes _`acc_`_.
```
Comment (dalg24) [inline P0009/P0009.md:2785]:
```suggestion
   * [3.3]{.pnum} Value-initializes _`acc_`_.
```

File: P0009/P0009.md [hunk 7]
```diff
@@ -2803,6 +2831,8 @@ constexpr mdspan(pointer p, const extents_type& ext);
 
    * [9.2]{.pnum} Direct-non-list-initializes _`map_`_ with `ext`.
 
+   * [2.3]{.pnum} Value-initializes _`acc_`_.
```
Comment (dalg24) [inline P0009/P0009.md:2834]:
```suggestion
   * [9.3]{.pnum} Value-initializes _`acc_`_.
```

File: P0009/P0009.md [hunk 8]
```diff
@@ -2839,24 +2869,23 @@ template<class OtherElementType, class OtherExtents,
                                 OtherLayoutPolicy, OtherAccessor>& other);
 ```
 
-[15]{.pnum} *Mandates:* 
-
-   * [15.1]{.pnum} `is_constructible_v<pointer, const OtherAccessor::pointer&>` is `true`;
-
-   * [15.2]{.pnum} `is_constructible_v<extents_type, OtherExtents>` is `true`.
-
 [16]{.pnum} *Constraints:*
```
Comment (dalg24) [inline P0009/P0009.md:2872]: Numbering is wrong below (skip 15)

File: P0009/P0009.md [hunk 9]
```diff
@@ -2785,6 +2811,8 @@ template<class SizeType, size_t N>
 
    * [6.2]{.pnum} Direct-non-list-initializes _`map_`_ with `extents_type(exts)`.
 
+   * [2.3]{.pnum} Value-initializes _`acc_`_.
```
Comment (dalg24) [inline P0009/P0009.md:2814]:
```suggestion
   * [6.3]{.pnum} Value-initializes _`acc_`_.
```

**Pull request merged**